packages/expo-updates/ios/EXUpdates/Update/EXUpdatesUpdate.m

```diff
@@ -58,45 +61,48 @@ + (instancetype)updateWithManifest:(NSDictionary *)manifest
                           response:(nullable NSURLResponse *)response
                             config:(EXUpdatesConfig *)config
                           database:(EXUpdatesDatabase *)database
+                             error:(NSError **)error
 {
-  if (config.usesLegacyManifest) {
+  if (![response isKindOfClass:[NSHTTPURLResponse class]]) {
+    *error = [NSError errorWithDomain:EXUpdatesUpdateErrorDomain
```

Comment:
sorry, one other thing i just noticed -- we should wrap just this one statement (not the `return`) in an additional `if (error)` check just in case we're given a null pointer.

Reply:
Can you explain what you are thinking a bit more? I thought we were assuming that the `error` was null here and assigning it a value?

<img width="655" alt="Screen Shot 2021-04-06 at 4 53 32 PM" src="https://user-images.githubusercontent.com/1220444/113791493-a0471180-96f8-11eb-88d5-c940f53f6c69.png">

Reply:
It's true that the `NSError *` is probably null, but note that we are assigning a value to `*error`, not `error`. `error` inside the method body is actually a `NSError **` (a pointer to a pointer) -- and it's possible we may want to pass nil for that if we don't care about the NSError object and only care about the method's return value.

Technically, the way you've written the method, the compiler would warn if you tried to pass `error:nil` when invoking this method, but that's a fairly common pattern so it would be good to handle it here anyway. And in fact, the method definition should really have `error:(NSError ** _Nullable)error` rather than just `NSError **` -- if you look at other examples of this pattern in the codebase they should look like that.

Let me know if that makes sense. This is one of those weird idiomatic Obj-C things -- a common pattern with NSError that you hardly see in any other context.

Reply:
Ah I see, that makes great sense. 

Out of curiosity, if we wanted to require an `error` always be passed would it be reasonable to pass just a single pointer and skip the null checks?

Reply:
No, if we wanted to require an `error` always be passed, we'd skip the `_Nullable` thing I suggested (basically write it just as you had written) and rely on the compiler warnings / human judgment to make sure `nil` was never passed.

The double pointer thing is basically a funky way to have two return values (the normal return value of the function, and the NSError) by allowing this method to make the _caller's_ pointer point to an entirely different object. If we were to pass a single pointer instead, we wouldn't be able to do that; the caller would have to initialize an NSError object and this method would have to mutate it (which isn't possible) rather than creating a new NSError.

Reply:
awesome. thanks.